npm/react/src/mount.ts

```diff
@@ -47,6 +47,6 @@ export function mount (jsx: React.ReactNode, options: MountOptions = {}, rerende
   return makeMountFn('mount', jsx, { ReactDom: ReactDOM, ...options }, rerenderKey, internalOptions)
 }
 
-export function unmount (options = { log: true }) {
```

Comment:
Should we deprecate and warn -> remove in v12? Just to make migration a bit smoother.

Reply:
I think marking as `deprecated` as Lachlan suggests makes sense.  As far as erroring or warning, my vote is to stay with erroring based on the assumption that someone calling `unmount` is either wanting to test the "unmount" behavior or perform some sort of cleanup.  Not erroring would most likely cause some sort of other error or problem that would be harder to track down.  I could be swayed otherwise though.

Reply:
Fine with a straight up deprecation and error *if we have a obvious alternative/migration strategy* - a straight up deprecation without a recommendation is like "your test is broken now, good luck fixing it" which is not a nice DX.

Reply:
@ZachJW34 @astone123 @warrensplayer I thought about this a bit more. Here's my recommendation:

1. Export this, but error. The error contains a link to the migration guide. [Added here](https://github.com/cypress-io/cypress/pull/24419/commits/e8af254b28368c07e270dab9c6ea0b7a92e12387).
2. Have a non exported `unmount` (I called it `internalUnmount`) we use for unmounting the component.
3. Provide a migration strategy. I added an example showing [how to unmount without `unmount` here](https://github.com/cypress-io/cypress/pull/24419/commits/b253b7f70100873bd3aacf62c65f4d68c49aee94#diff-67b4f03fe5a8e7deb9b4055c2d8e04b8bce37642571d970d474fd11e0e14094bR18-R30) and in my [migration docs](https://docs.google.com/document/d/1n4H0ADl0QNBWqbtGiKg4AzC0v8pQ8wZL0YLZeOCOg94/edit#).